examples/attempts_with_dlx_delayed_execution.rb

```diff
@@ -0,0 +1,70 @@
+# attempts_with_dead_letter_and_exponential_backoff.rb
+# ! check the examples/README.rdoc for information on starting your redis/rabbit !
+#
+# start it with ruby attempts_with_dead_letter_and_exponential_backoff.rb
+
+require "rubygems"
+require File.expand_path("../lib/beetle", File.dirname(__FILE__))
+
+require 'byebug'
+
+# set Beetle log level to info, less noisy than debug
+Beetle.config.logger.level = Logger::INFO
+
+# setup client with dead lettering enabled
+config = Beetle::Configuration.new
+config.dead_lettering_enabled = true
+config.dead_lettering_msg_ttl = 1000 # millis
+client = Beetle::Client.new(config)
+client.register_queue(:test)
+client.register_message(:test)
+client.register_message(:test_dead_letter)
+
+# purge the test queue
+client.purge(:test)
+
+# empty the dedup store
+client.deduplication_store.flushdb
+
+# setup our counter
+$completed = 0
+
+# store the start time
+$start_time = Time.now.to_f
+
+ReenqueueError = Class.new(StandardError)
+
+# declare a handler class for deferred/delayed message processing
+class Handler < Beetle::Handler
+  def process
+    logger.info "Attempts: #{message.attempts}, Defer: #{message.defer}, Processed after: #{Time.now.to_f - $start_time}"
+    logger.info message.data
+    $completed += 1
+  end
+end
+# for WORK QUEUE
+client.register_handler(:test, Handler)
+
+puts "Published 1 test message now to DEFERRED queue: #{Time.now.to_f}"
+client.publish(:test_dead_letter, {foo: :late}, headers: {defer: 6})
+client.publish(:test_dead_letter, {foo: :early}, headers: {defer: 3})
```

Comment:
this is a much better better solution than the one above (although the hack is not hidden in beetle's message.rb.)
- we can delay explicitly on publish
I use the headers param b/c it is piped through to the beetle message class in which I can evaluate the defer field and return early in order to delay the execution.
I use the dead_letter queue directly to ... mmh I forgot, perhaps it's not necessary anymore

Next step is something like:
```
client.publish_later(:test, data, 2.minutes.from_now)
```

Reply:
this solution can be quite easily properly integrated with a new option 'defer' that does not have to be in the headers field (that's just a hack to get it through for now)